src/main/java/org/broadinstitute/hdf5/HDF5File.java

```diff
@@ -477,21 +510,21 @@ private void checkIsOpen() {
     /**
      * Opens a HDF5 file given its access {@link OpenMode mode}.
      */
-    private static int open(final File file, final OpenMode mode) {
+    private static int open(final Path file, final OpenMode mode) {
         final int fileId;
         try {
             if (mode == OpenMode.CREATE) {
-                file.delete();
-                file.createNewFile();
+                Files.deleteIfExists(file);
+                Files.createFile(file);
             }
-            fileId = H5.H5Fopen(file.getAbsolutePath(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
+            fileId = H5.H5Fopen(file.toAbsolutePath().toString(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
```

Comment:
So this will work for `Path`s that are actually files on a local filesystem.  It will fail for a `Path` that is something more abstract, like a reference to an object in google's object store or a file stored in hdfs.  The main reason for us to switch to path was to support these sorts of non-file data storage containers.  I think if we want to support paths here, we'll need to A) check if the path is actually a local file and B) cache a non-local file somewhere on the local filesystem if it is not in order to provide a local file path.  

I think I would support a change that would do that with appropriate logging messages to help users understand what's happening, but just changing the type to Path without addressing the issue seems like a bad idea to me.  If the caller knows their path is actually a file they can always do the path-> file conversion themselves.